Packs/Carbon_Black_Enterprise_Response/Integrations/CarbonBlackResponseV2/CarbonBlackResponseV2_test.py

```diff
@@ -0,0 +1,310 @@
+        "test_path.exe|1|2097151|false",
+        [{'ProcessOpen_sub-type': 'handle open to process',
+          'cross-process_access_type': 'ProcessOpen',
+          'event_time': '2014-01-23 09:19:08.331',
+          'flagged_as_potential_tamper_attempt': 'false',
+          'requested_access_priviledges': '2097151',
+          'targeted_process_md5': '204f3f58212b3e422c90bd9691a2df28',
+          'targeted_process_path': 'test_path.exe',
+          'targeted_process_unique_id': '00000177-0000-0258-01cf-c209d9f1c431'}]
+    )
+]
+
+
+@pytest.mark.parametrize('data_str, expected', CROSSPROC_CASES)
+def test_crossproc(data_str, expected):
+    """
+        Given:
+            - A process event data containing crossproc field
+
+        When:
+            - formatting the data to correct format
+
+        Then:
+            - validating the new crossproc field contains json with correctly mapped data
+    """
+    from CarbonBlackResponseV2 import crossproc_complete
+
+    res = crossproc_complete(data_str).format()
+    assert res == expected
+
+
+@freeze_time("2021-03-14T13:34:14.758295Z")
+def test_fetch_incidents_first_fetch(mocker):
+    """
+        Given
+            fetch incidents command running for the first time.
+        When
+            mock the Client's http_request.
+        Then
+            validate fetch incidents command using the Client gets all relevant incidents
+    """
+    from CarbonBlackResponseV2 import fetch_incidents, Client
+    alerts = util_load_json('test_data/commands_test_data.json').get('fetch_incident_data')
+    client = Client(base_url="url", apitoken="api_key", use_ssl=True, use_proxy=False)
+    mocker.patch.object(Client, 'get_alerts', return_value=alerts)
+    first_fetch_time = '7 days'
+    _, incidents = fetch_incidents(client, last_run={}, first_fetch_time=first_fetch_time, max_results='3')
+    assert len(incidents) == 3
+    assert incidents[0].get('name') == 'Carbon Black EDR: 1 svchost.exe'
+
+
+@freeze_time("2021-03-16T13:34:14.758295Z")
+def test_fetch_incidents(mocker):
+    """
+        Given
+            fetch incidents command running for a second time.
+        When
+            mock the Client's http_request.
+        Then
+            validate fetch incidents command using the Client only returns new incidents
+    """
+    from CarbonBlackResponseV2 import fetch_incidents, Client
+    last_run = {'last_fetch': dateparser.parse('2021-03-12T14:13:20+00:00').timestamp()}
+    alerts = util_load_json('test_data/commands_test_data.json').get('fetch_incident_data')
+    client = Client(base_url="url", apitoken="api_key", use_ssl=True, use_proxy=False)
+    mocker.patch.object(Client, 'get_alerts', return_value=alerts)
+    first_fetch_time = '7 days'
+    _, incidents = fetch_incidents(client, last_run=last_run, first_fetch_time=first_fetch_time, max_results='3')
+    assert len(incidents) == 1
```

Comment:
please explain more in the When section,
I guess some of the incidents occurred time is previous to the `last_fetch`